etc/cp-ksql-server/questionnaire_app_event_observations.sql

```diff
@@ -0,0 +1,30 @@
+CREATE STREAM questionnaire_app_event (
+    projectId VARCHAR KEY,    -- 'KEY' means that this field is part of the kafka message key
+    userId VARCHAR KEY,
+    sourceId VARCHAR KEY,
+    questionnaireName VARCHAR,
+    eventType VARCHAR,
+    time DOUBLE,
+    metadata MAP<VARCHAR, VARCHAR>
+) WITH (
+    kafka_topic = 'questionnaire_app_event',
+    partitions = 3,
+    format = 'avro'
+);
+
+INSERT INTO observations
```

Comment:
Do you mean an autoincremented _id_ field?

Reply:
I have implemented a auto-incrementing 'id' field in data-dashboard-backend. The config in this PR has been updated accordingly.

Reply:
Oh sorry for the confusion i just meant the QUERY_ID -- https://docs.ksqldb.io/en/latest/developer-guide/ksqldb-reference/insert-into/#query_id (it is just a name for the INSERT query)

For the database tables, i think the primary keys you had previously were good. This is not that important in case of questionnaires, but when you have passive data such as fitbit there is usually high chance of having duplicate data, so this would reduce that in the database. Also would help with querying if using primary keys (as will be indexed by default).

Reply:
I added it like so:
```
INSERT INTO observations
WITH (QUERY_ID='questionnaire_app_event_observations')
SELECT
    q.projectId AS PROJECT,
```
and
```
INSERT INTO observations
WITH (QUERY_ID='questionnaire_response_observations')
SELECT 
```